File: python3/majority_element_ii.py
from typing import List

"""
Space   : O(n)
Time    : O(n)
"""


class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        ans = []
        n = len(nums)
        hashmap = {}
        
        for x in nums:
            if x in hashmap:
                hashmap[x] += 1
            else:
                hashmap[x] = 1
        
        for k, v in hashmap.items():
            if v > (n / 3):
                ans.append(k)

        return ans

File: golang/majority_element_ii.go
package main

// Space : O(n)
// Time  : O(n)

func majorityElementII(nums []int) []int {
	hashmap := map[int]int{}
	ans := []int{}
	n := len(nums)

	for _, x := range nums {
		hashmap[x]++
	}

	for k, v := range hashmap {
		if v > n/3 {
			ans = append(ans, k)
		}
	}

	return ans
}
